### PR TITLE
rephrase cbcs requirements based on CENC recommendation

### PR DESCRIPTION
**Pull request by cconcolato**

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -513,7 +513,7 @@ Common Encryption {#CommonEncryption}
 
 When the protected scheme <code>[=cenc=]</code> is used, samples SHALL be protected using subsample encryption and SHALL NOT use pattern encryption.
 
-When the protected scheme <code>[=cbcs=]</code> is used, samples SHALL be protected using subsample encryption and SHALL use pattern encryption. In particular, pattern encryption with <code>crypt_byte_block</code> = 1 and <code>skip_byte_block</code> = 0 SHALL be supported.
+When the protected scheme <code>[=cbcs=]</code> is used, samples SHALL be protected using subsample encryption and SHALL use pattern encryption. [[!CENC]] recommends that the Pattern Block length be 10, i.e. <code>crypt_byte_block + skip_byte_block = 10</code>. All combinations such that this is true SHALL be supported, in particular, <code>crypt_byte_block</code> = 10 and <code>skip_byte_block</code> = 0.
```
Comment (kqyang) [inline index.bs]: I still don't think it is a good idea to support non pattern encryption in pattern encryption scheme.

Anyway, this leaves the tail bytes, up to 159 bytes, unencrypted. Is that desirable?

Comment (cconcolato) [reply index.bs]: > Anyway, this leaves the tail bytes, up to 159 bytes, unencrypted. Is that desirable?

Can you clarify why you think so? CENC says:
> If the last Block pattern in a Subsample is incomplete, the partial pattern SHALL be followed until
truncated by the BytesOfProtectedData size and any partial crypt_byte_block SHALL
remain unencrypted.

I agree the wording is not perfect. I interpret as saying the last partial block (i.e. <16 bytes) remains unencrypted, and anyway partial blocks are not allowed.

Comment (kqyang) [reply index.bs]: I read the spec again. Yes, you are right. I mis-interpreted the spec previously.

**Comment by dwsinger** [comment]
Yes, IIRC in particular in cbcs it's not possible to encrypt a partial block; you have to pad it to a full-size, so what this is saying is exactly what you say; don't encrypt unless you have a full block in hand.

File: index.bs (hunk 1 above)
Comment (kqyang) [inline index.bs]: nit: s/in particular,/including/

**Pull request merged**